R/cmapplot.R

```diff
@@ -118,21 +118,40 @@ ggplot_size_conversion <- function(value, type = "bigpts") {
 }
 
 # Establish plotting constants in bigpts (1/72 of inch)
-cmapplot_globals$plot_constants <- list(
-  lwd_plotline = 3,
-  lwd_originline = 1.6, # This appears to be the minimum for variation between origin lines and other background lines
-  lwd_gridline = 0.3,
-  lwd_topline = 2,
-  margin_v1 = 5,
-  margin_v2 = 5,
-  margin_v3 = 5,
-  margin_v4 = 0,
-  margin_h1 = 2,
-  margin_h2 = 10,
-  margin_h3 = 10,
-  padding_plot = c(0,0,0,0),
-  leading_title = 0.93,
-  leading_caption = 1
+# [t] and [f] in comments indicate whether the constant is called in
+# [t]heme_cmap() and/or [f]inalize_plot().
+cmapplot_globals$consts <- list(
+  lwd_originline = 1.6, # [t, ] Width of origin lines. This appears to be the
+                        #        minimum for variation between origin lines and
+                        #        other background lines
+  lwd_gridline = 0.3,   # [t, ] Width of grid lines (non-origin)
+  lwd_plotline = 3,     # [ ,f] Width of line graph lines
+  lwd_topline = 2,      # [ ,f] Width of top line
+  margin_topline_t = 5, # [ ,f] Margin between top edge of image and top line
+  margin_title_t = 5,   # [ ,f] Margin between top line and title
+  margin_title_b = 5,   # [ ,f] Margin between title and caption if
+                        #        `caption_valign = "top"`
+  margin_caption_b = 5, # [ ,f] Margin between caption and bottom edge of image
+  margin_legend_t = 5,  # [ ,f] Margin between top line and plotbox
+  margin_legend_i = 5,  # [t,f] Vertical margin between legends (only applies
+                        #        to multilegend plots)
+  margin_legend_b = 10, # [t,f] Margin between legend and plot (within plotbox)
+  margin_plot_b = 5,    # [ ,f] Margin between plotbox and bottom edge of image
+  margin_title_l = 2,   # [ ,f] Margin between left edge of image and title/caption
+  margin_title_r = 10,  # [ ,f] Margin between title/caption and legend/plot
+                        #        (Both title_l and title_r are deducted from `titlewidth`)
+  margin_plot_r = 10,   # [ ,f] Margin between plotbox and right edge of image
+  padding_plot = c(0,10,0,0),
```

Comment:
I don't love how this removes the padding around plots in the RStudio "Plots" pane. Now my titles and axis text are all smashed up against the edge of the pane...

Reply:
lines 149 and 165 in `theme_cmap.R` were designed to address this on the top and the bottom in a situation with no titles. The idea was to create temporary spacing so it didn't look terrible at the `theme_cmap()` stage--but for this spacing to be stripped away once situated into the plotbox viewport during finalize. 

I clearly didn't think through the possible situations here, especially re the inclusion of title text. I'm open for suggestions on how to move forward. Is there an easy way to implement padding on all four sides of the larger ggplot object (regardless of the presence of legends, titles, axis titles, etc), that could get stripped away in finalize?

Reply:
I believe I have addressed this in commit ca042c1. @nmpeterson let me know if this solves your concerns?

Reply:
Looks much better, thanks